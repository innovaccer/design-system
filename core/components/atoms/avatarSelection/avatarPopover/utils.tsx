import * as React from 'react';

export const handleKeyDown = (
  event: React.KeyboardEvent,
  focusedOption: Element | undefined,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  setHighlightFirstItem?: React.Dispatch<React.SetStateAction<boolean>>,
  setHighlightLastItem?: React.Dispatch<React.SetStateAction<boolean>>,
  listRef?: any,
  withSearch?: boolean,
  setOpenPopover?: React.Dispatch<React.SetStateAction<boolean>>,
  triggerRef?: any
) => {
  switch (event.key) {
    case 'ArrowUp':
      event.preventDefault();
      navigateOptions('up', focusedOption, setFocusedOption, listRef, withSearch);
      break;
    case 'ArrowDown':
      event.preventDefault();
      navigateOptions('down', focusedOption, setFocusedOption, listRef, withSearch);
      break;
    case 'Enter':
      handleEnterKey(focusedOption);
      setHighlightLastItem?.(false);
      setHighlightFirstItem?.(false);
      break;
    case 'Escape':
      setOpenPopover?.(false);
      triggerRef.current.focus();
      setFocusedOption?.(undefined);
      break;
    default:
      break;
  }
};

const handleEnterKey = (focusedOption: Element | undefined) => {
  (focusedOption as HTMLElement)?.click();
};

const navigateOptions = (
  direction: string,
  focusedOption: Element | undefined,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  listRef?: any,
  withSearch?: boolean
) => {
  const listItems = listRef?.current.querySelectorAll('[data-test="DesignSystem-Listbox-ItemWrapper"]');
  let index = Array.from(listItems).findIndex((item) => {
    return item == focusedOption;
  });

  if (index === -1) {
    index = direction === 'up' ? listItems.length - 1 : 0;
  } else if (
    (withSearch && index === 0 && direction === 'up') ||
    (withSearch && index === listItems.length - 1 && direction === 'down')
  ) {
    const searchInput = listRef.current.querySelector('[data-test="DesignSystem-AvatarSelection--Input"]');
    searchInput.focus();
    setFocusedOption && setFocusedOption(searchInput);
  } else {
    index = direction === 'up' ? (index - 1 + listItems.length) % listItems.length : (index + 1) % listItems.length;

    const targetOption = listItems[index];

    (targetOption as HTMLElement).focus();
    setFocusedOption && setFocusedOption(targetOption);
    targetOption.scrollIntoView({ block: 'center' });
  }
};

export const handleInputKeyDown = (
  event: React.KeyboardEvent,
  listRef: any,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  setOpenPopover?: React.Dispatch<React.SetStateAction<boolean>>,
  triggerRef?: any
) => {
  const listItems = listRef.current?.querySelectorAll('[data-test="DesignSystem-Listbox-ItemWrapper"]');
  let targetOption;

  switch (event.key) {
    case 'ArrowUp':
      event.preventDefault();
      targetOption = listItems[listItems.length - 1];
      break;
    case 'ArrowDown':
      event.preventDefault();
      targetOption = listItems[0];
      break;
    case 'Escape':
      setOpenPopover?.(false);
      triggerRef.current.focus();
      setFocusedOption?.(undefined);
      break;
    default:
      break;
  }

  (targetOption as HTMLElement)?.focus();
  targetOption?.scrollIntoView?.({ block: 'center' });
  setFocusedOption && setFocusedOption(targetOption);
};
