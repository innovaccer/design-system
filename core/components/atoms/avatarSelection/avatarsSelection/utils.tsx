import * as React from 'react';

export const handleKeyDown = (
  event: React.KeyboardEvent,
  setOpenPopover?: React.Dispatch<React.SetStateAction<boolean>>,
  setHighlightFirstItem?: React.Dispatch<React.SetStateAction<boolean>>,
  setHighlightLastItem?: React.Dispatch<React.SetStateAction<boolean>>
) => {
  switch (event.key) {
    case 'Enter':
      setOpenPopover?.(true);
      setHighlightFirstItem?.(true);
      break;
    case 'ArrowDown':
      setOpenPopover?.(true);
      setHighlightFirstItem?.(true);
      break;
    case 'ArrowUp':
      setOpenPopover?.(true);
      setHighlightLastItem?.(true);
      break;
    default:
      break;
  }
};

export const focusListItem = (
  position: string,
  setFocusedOption?: React.Dispatch<React.SetStateAction<HTMLElement | undefined>>,
  listRef?: any,
  withSearch?: boolean
) => {
  if (withSearch && position === 'down') {
    const searchInput = listRef.current?.querySelectorAll('[data-test="DesignSystem-AvatarSelection--Input"]');
    searchInput && searchInput[0]?.focus();
    setFocusedOption?.(searchInput);
    return;
  }

  const listItems = listRef.current?.querySelectorAll('[data-test="DesignSystem-Listbox-ItemWrapper"]');
  let targetOption;

  if (position === 'down') {
    targetOption = listItems?.[0];
  } else {
    targetOption = listItems?.[listItems.length - 1];
  }
  (targetOption as HTMLElement)?.focus();
  targetOption?.scrollIntoView?.({ block: 'center' });
  setFocusedOption && setFocusedOption(targetOption);
};
