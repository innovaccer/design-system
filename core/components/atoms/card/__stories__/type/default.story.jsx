import { Card } from '@/index';

// CSF format story
export const defaultCard = () => {
  return (
    <Card>
      <div className="p-8" />
    </Card>
  );
};

export default {
  title: 'Components/Card/Type/Default Card',
  component: Card,
};
