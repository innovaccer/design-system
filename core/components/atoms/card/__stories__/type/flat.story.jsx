import { Card } from '@/index';

// CSF format story
export const Flat = () => {
  return (
    <Card shadow="none">
      <div className="p-8" />
    </Card>
  );
};

export default {
  title: 'Components/Card/Type/Flat',
  component: Card,
};
