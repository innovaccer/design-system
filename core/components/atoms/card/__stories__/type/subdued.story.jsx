import { Card, CardSubdued } from '@/index';

// CSF format story
export const Subdued = () => {
  return (
    <CardSubdued>
      <div className="p-8" />
    </CardSubdued>
  );
};

export default {
  title: 'Components/Card/Type/Subdued',
  component: Card,
  subcomponents: { CardSubdued },
  parameters: {
    docs: {
      docPage: {
        title: 'CardSubdued',
        description: 'A component to highlight section of a card.',
      },
    },
  },
};
