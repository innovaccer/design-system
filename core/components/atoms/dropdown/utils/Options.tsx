export let dropdownOptions: any[] = [];

for (let i = 1; i <= 50; i++) {
  dropdownOptions.push({
    label: `Option${i}`,
    value: `Option${i}`,
  });
}

export const selectedStoryOptions: any[] = [];

for (let i = 1; i <= 10; i++) {
  selectedStoryOptions.push({
    label: `Option${i}`,
    value: `Option${i}`,
    selected: i === 2,
  });
}

export const multiSelectedStoryOptions: any[] = [];

for (let i = 1; i <= 10; i++) {
  multiSelectedStoryOptions.push({
    label: `Option${i}`,
    value: `Option${i}`,
    selected: i === 2 || i === 7,
  });
}

export const storyOptions = dropdownOptions.slice(0, 10);

export const storySections = { 0: 'subheading', 5: 'subheading' };

export const subInfoItems = [
  {
    label: 'Option1',
    value: 'Option1',
    subInfo: 'subInfo'
  },
  {
    label: 'Option2',
    value: 'Option2',
    subInfo: 'subInfo'
  },
  {
    label: 'Option3',
    value: 'Option3',
    subInfo: 'subInfo',
  },
];

export const iconItems = [
  {
    label: 'Option1',
    value: 'Option1',
    icon: 'events'
  },
  {
    label: 'Option2',
    value: 'Option2',
    icon: 'events',
  },
  {
    label: 'Option3',
    value: 'Option3',
    icon: 'events',
  },
];
