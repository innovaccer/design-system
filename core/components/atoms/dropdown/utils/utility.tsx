const getSearchedOptions = (options: any, searchTerm: string) => {
  const result = options.filter((option: any) => option.label.toLowerCase().includes(searchTerm.toLowerCase()));
  return result;
};

export const getOptions = (offset: number, limit: number, searchTerm: string, options: any) => {
  const searchedOptions = searchTerm ? getSearchedOptions(options, searchTerm) : options;
  return new Promise(resolve => {
    resolve({
      offset,
      slicedOptions: searchedOptions.slice(offset, offset + limit),
      options: searchedOptions,
    });
  });
};

export const getValuesFromSelectedObj = (selectedArray: any[] = []) => {
  const selectedValues: any[] = [];

  selectedArray.forEach((selectedObj: any) => {
    const { value } = selectedObj;
    selectedValues.push(value);
  });

  return selectedValues;
};

export const getIndexesFromSelectedObj = (selectedArray: any[] = []) => {
  return (Array.from(Array(selectedArray.length).keys()));
};

export const getValuesFromInd = (selectedArray: number[], options: any[]) => {
  const result: any[] = [];
  selectedArray.forEach(ind => {
    result.push(options[ind].value);
  });

  return result;
};

export const getLabelsFromInd = (selectedArray: number[], options: any[]) => {
  const result: string[] = [];
  selectedArray.forEach(ind => {
    result.push(options[ind].label);
  });

  return result;
};
