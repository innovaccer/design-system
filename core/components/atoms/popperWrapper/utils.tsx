// Helper function to convert CSS class names to style objects
export const createStyleFromClass = (className: string) => {
  if (!className) return {};

  // Create a temporary element to get computed styles
  const tempElement = document.createElement('div');
  tempElement.className = className;
  document.body.appendChild(tempElement);

  const computedStyle = window.getComputedStyle(tempElement);
  const style: React.CSSProperties = {};

  // Extract relevant animation properties
  const animation = computedStyle.animation;
  const animationDuration = computedStyle.animationDuration;
  const animationTimingFunction = computedStyle.animationTimingFunction;
  const animationFillMode = computedStyle.animationFillMode;
  const opacity = computedStyle.opacity;
  const transform = computedStyle.transform;

  if (animation && animation !== 'none') {
    style.animation = animation;
  }
  if (animationDuration && animationDuration !== '0s') {
    style.animationDuration = animationDuration;
  }
  if (animationTimingFunction && animationTimingFunction !== 'ease') {
    style.animationTimingFunction = animationTimingFunction;
  }
  if (animationFillMode && animationFillMode !== 'none') {
    style.animationFillMode = animationFillMode;
  }
  if (opacity && opacity !== '1') {
    style.opacity = opacity;
  }
  if (transform && transform !== 'none') {
    style.transform = transform;
  }

  document.body.removeChild(tempElement);
  return style;
};
