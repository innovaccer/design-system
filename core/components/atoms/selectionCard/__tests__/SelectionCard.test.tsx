import { fireEvent, render } from '@testing-library/react';
import { testHelper, filterUndefined, valueHelper, testMessageHelper } from '@/utils/testHelper';
import { SelectionCard } from '@/index';
import { SelectionCardProps as Props } from '@/index.type';

const BooleanValue = [true, false];
const FunctionValue = jest.fn();
const StringValue = 'Selection Card';
const name = 'card1';

const SingleSelect = () => {
  const { isCardSelected, updateCardSelection } = SelectionCard.useSingleSelect();

  return (
    <div>
      <SelectionCard
        id="item1"
        data-test="Selection-card-1"
        onClick={() => updateCardSelection('item1')}
        selected={isCardSelected('item1')}
      >
        Single Selection Item 1
      </SelectionCard>

      <SelectionCard
        id="item2"
        data-test="Selection-card-2"
        onClick={() => updateCardSelection('item2')}
        selected={isCardSelected('item2')}
      >
        Single Selection Item 2
      </SelectionCard>

      <SelectionCard
        id="item3"
        data-test="Selection-card-3"
        disabled={true}
        onClick={() => updateCardSelection('item3')}
        selected={isCardSelected('item3')}
      >
        Single Selection Item 3
      </SelectionCard>
    </div>
  );
};

const MultiSelect = () => {
  const { isCardSelected, updateCardSelection } = SelectionCard.useMultiSelect();

  return (
    <div>
      <SelectionCard
        id="item1"
        cardValue={{ title: 'Item 1' }}
        data-test="Selection-card-1"
        onClick={() => updateCardSelection('item1', { title: 'Item 1' })}
        selected={isCardSelected('item1')}
      >
        Multi Selection
      </SelectionCard>

      <SelectionCard
        id="item2"
        cardValue={{ title: 'Item 2' }}
        data-test="Selection-card-2"
        onClick={() => updateCardSelection('item2', { title: 'Item 2' })}
        selected={isCardSelected('item2')}
      >
        Multi Selection
      </SelectionCard>

      <SelectionCard
        id="item3"
        data-test="Selection-card-3"
        disabled={true}
        onClick={() => updateCardSelection('item3')}
        selected={isCardSelected('item3')}
      >
        Single Selection Item 3
      </SelectionCard>
    </div>
  );
};

describe('selection card item component snapshot', () => {
  const mapper: Record<string, any> = {
    disabled: valueHelper(BooleanValue, { required: false, iterate: true }),
    selected: valueHelper(BooleanValue, { required: false, iterate: true }),
    children: valueHelper(StringValue, { required: true }),
    onClick: valueHelper(FunctionValue, { required: false }),
  };

  const testFunc = (props: Record<string, any>): void => {
    const attr = filterUndefined(props) as Props;

    it(testMessageHelper(attr), () => {
      const { baseElement } = render(<SelectionCard {...attr} />);
      expect(baseElement).toMatchSnapshot();
    });
  };

  testHelper(mapper, testFunc);
});

describe('selection card component tests', () => {
  it('check for render item', () => {
    const { getByTestId } = render(<SelectionCard id={name}>{StringValue}</SelectionCard>);
    const element = getByTestId('DesignSystem-SelectionCard');
    expect(element).toBeInTheDocument();
    expect(element).toHaveClass('Selection-card');
    expect(element).toHaveClass('Selection-card--default');
    expect(element).toHaveTextContent(StringValue);
  });
});

describe('selection card item overlay component tests', () => {
  it('check for overlay item', () => {
    const { getByTestId } = render(<SelectionCard id={name}>{StringValue}</SelectionCard>);
    const element = getByTestId('DesignSystem-SelectionCard-Overlay');
    expect(element).toBeInTheDocument();
    expect(element).toHaveClass('Selection-card-overlay');
  });

  it('check for overlay item z-index', () => {
    const { getByTestId } = render(
      <SelectionCard id={name} overlayZIndex={100}>
        {StringValue}
      </SelectionCard>
    );
    const element = getByTestId('DesignSystem-SelectionCard-Overlay');
    expect(element).toHaveStyle({ zIndex: 100 });
  });
});

describe('selection card disabled item component tests', () => {
  it('check for disabled class for unselected item', () => {
    const { getByTestId } = render(
      <SelectionCard id={name} disabled={true}>
        {StringValue}
      </SelectionCard>
    );
    const element = getByTestId('DesignSystem-SelectionCard');
    expect(element).toHaveClass('Selection-card--disabled');
    expect(element).toHaveClass('Selection-card--default-disabled');
  });

  it('check for disabled class for selected item', () => {
    const { getByTestId } = render(
      <SelectionCard id={name} disabled={true} selected={true}>
        {StringValue}
      </SelectionCard>
    );
    const element = getByTestId('DesignSystem-SelectionCard');
    expect(element).toHaveClass('Selection-card--disabled');
    expect(element).toHaveClass('Selection-card--selected-disabled');
  });

  it('check for onClick event for disabled item', () => {
    const { getByTestId } = render(
      <SelectionCard id={name} onClick={FunctionValue} disabled={true} selected={true}>
        {StringValue}
      </SelectionCard>
    );
    const element = getByTestId('DesignSystem-SelectionCard');
    fireEvent.click(element);
    expect(FunctionValue).not.toHaveBeenCalled();
  });
});

describe('selection card component classes', () => {
  it('check for selection classes', () => {
    const { getByTestId } = render(
      <SelectionCard id={name} selected={true}>
        {StringValue}
      </SelectionCard>
    );
    const element = getByTestId('DesignSystem-SelectionCard');
    expect(element).toHaveClass('Selection-card--selected');
  });

  it('check for custom class in selection card component', () => {
    const { getByTestId } = render(
      <SelectionCard id={name} className="custom-class">
        {StringValue}
      </SelectionCard>
    );
    expect(getByTestId('DesignSystem-SelectionCard')).toHaveClass('custom-class');
  });
});

describe('selection card component event handlers', () => {
  it('check for click event handler', () => {
    const { getByTestId } = render(
      <SelectionCard id={name} onClick={FunctionValue}>
        {StringValue}
      </SelectionCard>
    );
    const element = getByTestId('DesignSystem-SelectionCard');
    fireEvent.click(element);
    expect(FunctionValue).toHaveBeenCalled();
  });
});

describe('selection card component hooks test', () => {
  it('test for single select hook', () => {
    const { getByTestId } = render(<SingleSelect />);
    const element1 = getByTestId('Selection-card-1');
    fireEvent.click(element1);
    expect(element1).toHaveClass('Selection-card--selected');

    const element2 = getByTestId('Selection-card-2');
    fireEvent.click(element2);
    expect(element2).toHaveClass('Selection-card--selected');
    expect(element1).not.toHaveClass('Selection-card--selected');

    fireEvent.click(element2);
    expect(element2).not.toHaveClass('Selection-card--selected');
    const element3 = getByTestId('Selection-card-3');
    fireEvent.click(element2);

    fireEvent.click(element3);
    expect(element3).not.toHaveClass('Selection-card--selected');
    expect(element2).toHaveClass('Selection-card--selected');
  });

  it('test for multi select hook', () => {
    const { getByTestId } = render(<MultiSelect />);
    const element1 = getByTestId('Selection-card-1');
    fireEvent.click(element1);
    expect(element1).toHaveClass('Selection-card--selected');

    const element2 = getByTestId('Selection-card-2');
    fireEvent.click(element2);
    expect(element2).toHaveClass('Selection-card--selected');

    fireEvent.click(element1);
    expect(element1).not.toHaveClass('Selection-card--selected');

    const element3 = getByTestId('Selection-card-3');
    fireEvent.click(element3);
    expect(element3).not.toHaveClass('Selection-card--selected');

    expect(element2).toHaveClass('Selection-card--selected');
  });
});
