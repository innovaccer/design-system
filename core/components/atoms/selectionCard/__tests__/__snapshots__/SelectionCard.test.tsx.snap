// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`selection card item component snapshot 
  children: "Selection Card", disabled: false, onClick: [null,"[Function]"]
 1`] = `
<body>
  <div>
    <div
      class="Selection-card Selection-card--default"
      data-test="DesignSystem-SelectionCard"
      role="checkbox"
      tabindex="0"
    >
      <div
        class="Selection-card-overlay"
        data-test="DesignSystem-SelectionCard-Overlay"
        style="z-index: 2;"
      />
      Selection Card
    </div>
  </div>
</body>
`;

exports[`selection card item component snapshot 
  children: "Selection Card", disabled: false, selected: false, onClick: [null,"[Function]"]
 1`] = `
<body>
  <div>
    <div
      aria-checked="false"
      class="Selection-card Selection-card--default"
      data-test="DesignSystem-SelectionCard"
      role="checkbox"
      tabindex="0"
    >
      <div
        class="Selection-card-overlay"
        data-test="DesignSystem-SelectionCard-Overlay"
        style="z-index: 2;"
      />
      Selection Card
    </div>
  </div>
</body>
`;

exports[`selection card item component snapshot 
  children: "Selection Card", disabled: false, selected: true, onClick: [null,"[Function]"]
 1`] = `
<body>
  <div>
    <div
      aria-checked="true"
      class="Selection-card Selection-card--default Selection-card--selected"
      data-test="DesignSystem-SelectionCard"
      role="checkbox"
      tabindex="0"
    >
      <div
        class="Selection-card-overlay"
        data-test="DesignSystem-SelectionCard-Overlay"
        style="z-index: 2;"
      />
      Selection Card
    </div>
  </div>
</body>
`;

exports[`selection card item component snapshot 
  children: "Selection Card", disabled: true, onClick: [null,"[Function]"]
 1`] = `
<body>
  <div>
    <div
      class="Selection-card Selection-card--disabled Selection-card--default-disabled"
      data-test="DesignSystem-SelectionCard"
      role="checkbox"
      tabindex="-1"
    >
      <div
        class="Selection-card-overlay"
        data-test="DesignSystem-SelectionCard-Overlay"
        style="z-index: 2;"
      />
      Selection Card
    </div>
  </div>
</body>
`;

exports[`selection card item component snapshot 
  children: "Selection Card", disabled: true, selected: false, onClick: [null,"[Function]"]
 1`] = `
<body>
  <div>
    <div
      aria-checked="false"
      class="Selection-card Selection-card--disabled Selection-card--default-disabled"
      data-test="DesignSystem-SelectionCard"
      role="checkbox"
      tabindex="-1"
    >
      <div
        class="Selection-card-overlay"
        data-test="DesignSystem-SelectionCard-Overlay"
        style="z-index: 2;"
      />
      Selection Card
    </div>
  </div>
</body>
`;

exports[`selection card item component snapshot 
  children: "Selection Card", disabled: true, selected: true, onClick: [null,"[Function]"]
 1`] = `
<body>
  <div>
    <div
      aria-checked="true"
      class="Selection-card Selection-card--disabled Selection-card--selected-disabled"
      data-test="DesignSystem-SelectionCard"
      role="checkbox"
      tabindex="-1"
    >
      <div
        class="Selection-card-overlay"
        data-test="DesignSystem-SelectionCard-Overlay"
        style="z-index: 2;"
      />
      Selection Card
    </div>
  </div>
</body>
`;

exports[`selection card item component snapshot 
  children: "Selection Card", onClick: [null,"[Function]"]
 1`] = `
<body>
  <div>
    <div
      class="Selection-card Selection-card--default"
      data-test="DesignSystem-SelectionCard"
      role="checkbox"
      tabindex="0"
    >
      <div
        class="Selection-card-overlay"
        data-test="DesignSystem-SelectionCard-Overlay"
        style="z-index: 2;"
      />
      Selection Card
    </div>
  </div>
</body>
`;

exports[`selection card item component snapshot 
  children: "Selection Card", selected: false, onClick: [null,"[Function]"]
 1`] = `
<body>
  <div>
    <div
      aria-checked="false"
      class="Selection-card Selection-card--default"
      data-test="DesignSystem-SelectionCard"
      role="checkbox"
      tabindex="0"
    >
      <div
        class="Selection-card-overlay"
        data-test="DesignSystem-SelectionCard-Overlay"
        style="z-index: 2;"
      />
      Selection Card
    </div>
  </div>
</body>
`;

exports[`selection card item component snapshot 
  children: "Selection Card", selected: true, onClick: [null,"[Function]"]
 1`] = `
<body>
  <div>
    <div
      aria-checked="true"
      class="Selection-card Selection-card--default Selection-card--selected"
      data-test="DesignSystem-SelectionCard"
      role="checkbox"
      tabindex="0"
    >
      <div
        class="Selection-card-overlay"
        data-test="DesignSystem-SelectionCard-Overlay"
        style="z-index: 2;"
      />
      Selection Card
    </div>
  </div>
</body>
`;
